### ISSUE TITLE
A special case that requires filter over TokenList id

### ISSUE THREAD

**Issue by Vimos** [EmilStenstrom/conllu#47]
For the following example,
```
# sent_id = reviews-002288-0001
# newpar id = reviews-002288-p0001
# text = It's well cool. :)
1-2	It's	_	_	_	_	_	_	_	_
1	It	it	PRON	PRP	Case=Nom|Gender=Neut|Number=Sing|Person=3|PronType=Prs	4	nsubj	4:nsubj	_
2	's	be	AUX	VBZ	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	4	cop	4:cop	_
3	well	well	ADV	RB	Degree=Pos	4	advmod	4:advmod	_
4	cool	cool	ADJ	JJ	Degree=Pos	0	root	0:root	SpaceAfter=No
5	.	.	PUNCT	.	_	4	punct	4:punct	_
6	:)	:)	SYM	NFP	_	4	discourse	4:discourse	_
```
The parsed result will be 
```
TokenList<It's, It, 's, well, cool, ., :)>

[{'id': (1, '-', 2),
  'form': "It's",
  'lemma': '_',
  'upos': '_',
  'xpos': None,
  'feats': None,
  'head': None,
  'deprel': '_',
  'deps': None,
  'misc': None},
 {'id': 1,
  'form': 'It',
  'lemma': 'it',
  'upos': 'PRON',
  'xpos': 'PRP',
  'feats': {'Case': 'Nom',
   'Gender': 'Neut',
   'Number': 'Sing',
   'Person': '3',
   'PronType': 'Prs'},
  'head': 4,
  'deprel': 'nsubj',
  'deps': [('nsubj', 4)],
  'misc': None},
]
```

The parsing has no problem. However, in some cases, one may want to use those `id` that is integer.

How about make `TokenList` to support filter those with only integer id?
Like this 
```python
sentence.filter(id=lambda x: type(x) is int)
```

**Comment by EmilStenstrom** [comment]
@Vimos That's a great idea, I'm 👍 to adding it to conllu.

I don't have much bandwidth to work on it right now, so if you're able to make a PR (preferably with passing tests! :) ) that would speed things up a bit!

**Comment by Vimos** [comment]
Yes. Sure!

I will make a PR soon! This modification will support `lambda` filters for all fields, I suppose.

```python
    def test_lambda_filtering(self):
        tokenlist = TokenList([
            {'id': (1, '-', 2),
             'form': "It's",
             'lemma': '_',
             'upos': '_',
             'xpos': None,
             'feats': None,
             'head': None,
             'deprel': '_',
             'deps': None,
             'misc': None},
            {'id': 1,
             'form': 'It',
             'lemma': 'it',
             'upos': 'PRON',
             'xpos': 'PRP',
             'feats': {'Case': 'Nom',
                       'Gender': 'Neut',
                       'Number': 'Sing',
                       'Person': '3',
                       'PronType': 'Prs'},
             'head': 4,
             'deprel': 'nsubj',
             'deps': [('nsubj', 4)],
             'misc': None},
            {'id': 2,
             'form': "'s",
             'lemma': 'be',
             'upos': 'AUX',
             'xpos': 'VBZ',
             'feats': {'Mood': 'Ind',
                       'Number': 'Sing',
                       'Person': '3',
                       'Tense': 'Pres',
                       'VerbForm': 'Fin'},
             'head': 4,
             'deprel': 'cop',
             'deps': [('cop', 4)],
             'misc': None}
        ])
        self.assertEqual(
            tokenlist.filter(id=lambda x: type(x) is int),
            tokenlist[1:]
        )
        self.assertEqual(
            tokenlist.filter(upos=lambda x: x.startswith('P')),
            TokenList([tokenlist[1]])
        )
```

**Comment by EmilStenstrom** [comment]
@Vimos Great! I think you can minimize the test so only the fields you actually test against are part of the test data. Otherwise looks great!

**Comment by Vimos** [comment]
I grouped them into `lambda_basic` and `lambda_deep`, the test cases are also simplified.

**Comment by EmilStenstrom** [comment]
I added a brief section about filtering by lambda in the README, and just released conllu==4.3 that supports this. Thanks for your contribution!
